### PR TITLE
Read Menu object from .yml file

### LINKED ISSUE #27: Implement yml menu reader
1. Learn the basics of yaml
2. Write a helper function that reads the yml text from a .yml file
3. Design menu class. Will likely include delivery cost (int), instructions to contact restaurant (text string), dishes (dict), modifiers (dict)
4. Main goal is to write a function that takes the yaml text (not file name) and returns a menu object

### PR DESCRIPTION
**Pull request by rushilramesh**
- `ymlFileToDict` reads raw text from file and node library converts it to a dict
- `getMenu` returns an Menu object

### CODE REVIEW

File: menu/reader.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Import Dependencies
+const fs = require("fs");
+const YAML = require("js-yaml");
+
+class Menu {
+    constructor(text) {
+        this.delivery = text.Delivery;
+        this.instructions = text?.Instructions; // If instructions provided
```
Comment (nicktohzyu) [inline menu/reader.js:8]: is the `?` supposed to be there?

File: menu/reader.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.dishes = text.Menu;
+        this.modifiers = text.Modifiers
+    }
+
+}
+
+// Convert .yml file to object
+const ymlFileToDict = (filename) => {
```
Comment (nicktohzyu) [inline menu/reader.js:16]: don't use lambda

File: menu/reader.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    try {
+        const raw = fs.readFileSync(filename);
```
Comment (nicktohzyu) [inline menu/reader.js:18]: could you look into what's the return type of readFileSync (is it a utf-8 string?)

File: menu/reader.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        const data = YAML.load(raw)
+        return data
+    } catch (e) {
+        console.log(e)
```
Comment (nicktohzyu) [inline menu/reader.js:22]: distinguish the file read error from a yaml parsing error

**Comment by nicktohzyu** [comment]
Usually i'd ask you to separate the file reading from yaml parsing, but I think it's sufficiently simple here that there's no need for now. But do update the error catching to be more meaningful

File: menu/reader.js (hunk 1 above)
Comment (rushilramesh) [reply menu/reader.js:8]: The '?' is just to indicate it's an optional field as the current menu.yml file does not contain it, but I can remove it doesn't affect the object

File: menu/reader.js (hunk 2 above)
Comment (rushilramesh) [reply menu/reader.js:16]: Oki!

File: menu/reader.js (hunk 4 above)
Comment (rushilramesh) [reply menu/reader.js:22]: Will do

**Pull request closed without merging**